src/processout/applepay.ts

```diff
@@ -5,280 +5,312 @@
 
-                }, function(req: XMLHttpRequest, e: Event, errorCode: ApiRequestError): void {
-                    t.onerror(new Exception(errorCode));
-                    t.session.abort();
-                });
-            };
-            // As well as the other ones
-            this.session.oncancel = this.onCancelHandler.bind(this);
-            this.session.onshippingcontactselected = this.onShippingContactSelectedHandler.bind(this);
-            this.session.onshippingmethodselected = this.onShippingMethodSelectedHandler.bind(this);
-        }
+    /**
+     * ApplePay constructor
+     * @param {ProcessOut} instance
+     * @param {ApplePayPaymentRequest} req
+     */
+    constructor(instance: ProcessOut, req: ApplePayPaymentRequest) {
+      this.instance = instance;
 
-        /**
-         * SetHandlers sets the success and error handlers
-         * @param {(e: any) => void} onsuccess
-         * @param {(err: Exception) => void} onerror
-         * @return {void}
-         */
-        protected setHandlers(onsuccess: (e: any) => void, 
-            onerror?: (err: Exception) => void): void {
-            
-            if (!onsuccess)
-                throw new Exception("applepay.no-success-handler");
-            this.onsuccess = onsuccess;
-            this.onerror = function(err: Exception): void {
-                if (onerror) onerror(err);   
-            };
-        }
+      // We want to set sensible defaults on the Apple Pay request if
+      // it wasn't previously set by the merchant
+      if (!req.merchantCapabilities || !req.merchantCapabilities.length)
+        req.merchantCapabilities = ["supports3DS"];
+      if (!req.supportedNetworks || !req.supportedNetworks.length)
+        req.supportedNetworks = ["amex", "discover", "masterCard", "visa"];
+      this.request = req;
 
-        /**
-         * Tokenize handles the Apple Pay payment sheet
-         * @param {any} data
-         * @param {(e: any) => void} onsuccess
-         * @param {(err: Exception) => void} onerror
-         * @return {void}
-         */
-        public tokenize(data: any, onsuccess: (e: any) => void, 
-            onerror?: (err: Exception) => void): void {
-            
-            this.setHandlers(onsuccess, onerror);
-            this.session.begin();
-        }
+      // Let's now create the session so we can wrap it in our
+      // Apple Pay class
+      this.session = new ApplePaySession(1, this.request);
+      // Hook the session events we need
+      var t = this;
+      this.session.onvalidatemerchant = function (event: any): void {
+        t.instance.apiRequest(
+          "post",
+          t.instance.endpoint("api", "/applepay/sessions"),
+          {
+            session_url: event.validationURL,
+            domain_name: window.location.hostname,
+            merchant_certificate_id: t.request.merchantCertificateId,
+            applepay_mid: t.request.merchantApplePayCertificateId,
```

Comment:
Is this the only real change in this PR? It is quite hard to identify among the format change even filter out the whitespace.

Reply:
It would be nice to have 2 commits, separate the format changes from the actual code changes.

Reply:
Yeah you right, here are the 3 changes: 

- https://github.com/processout/processout.js/pull/78/files#diff-9a80fcfc5f93fa79d9070aa264a6c84032056e11cbe3ae4bbead6ca8a80ca167R131
- https://github.com/processout/processout.js/pull/78/files#diff-9a80fcfc5f93fa79d9070aa264a6c84032056e11cbe3ae4bbead6ca8a80ca167R108
- https://github.com/processout/processout.js/pull/78/files#diff-3e255e8bd53882cb8469283995b924befca568fc11e5c8a0cabcccde6b360ee3R17